### PR TITLE
TK-06149 hdk references

### PR DESCRIPTION
**Pull request by thedavidmeister**

### CODE REVIEW

File: crates/hdk/README.md [hunk 1]
```diff
@@ -556,7 +556,7 @@ they are finalized on the local source chain or broadcast to the DHT network.
 In addition to the patterns we demonstrated above for externs and callbacks, we
 also need to introduce the `host_call()!` and `host_args!()` macros.
 
-The `host_call!()` macro works very similarly to the `ret!()` macro in that it
+The `host_call()` macro works very similarly to the `ret!()` macro in that it
```
Comment (steveeJ) [inline crates/hdk/README.md:559]:
```suggestion
The `host_call()` function works very similarly to the `ret!()` macro in that it
```

File: crates/hdk/README.md [hunk 2]
```diff
@@ -572,7 +572,7 @@ always call it before anything else. Note that `host_args!()` will short circuit
 and return early with an error similar to the `?` operator if deserialization
 fails on the guest.
 
-Note that both the `host_call!()` and `host_args!()` macros rely on the guest to
+Note that both the `host_call()` and `host_args!()` macros rely on the guest to
```
Comment (steveeJ) [inline crates/hdk/README.md:575]:
```suggestion
Note that both the `host_call()` function and the `host_args!()` macros rely on the guest to
```

File: crates/hdk/README.md [hunk 3]
```diff
@@ -831,9 +831,9 @@ struct Png([u8]);
 // no need for special keywords on the function
 // the input args and return values are all native Rust types, not pointers
 // we can use `Result` and `?`
-// the only wasm-ey thing here is the `host_call!()` macro
+// the only wasm-ey thing here is the `host_call()` macro
 fn _save_image(png: Png) -> Result<CreateOutput, String> {
- Ok(host_call!(
+ Ok(host_call(
   __create,
   CreateInput::new(
```
Comment (steveeJ) [inline crates/hdk/README.md:838]:
```suggestion
  &CreateInput::new(
```

**Review by steveeJ — COMMENTED** [review]
Some now outdated references to `host_call` as a macro.

Can we get red of the many `.into_inner()`s somehow?

**Comment by thedavidmeister** [comment]
> Some now outdated references to `host_call` as a macro.
> 
> Can we get red of the many `.into_inner()`s somehow?

@steveeJ yeah i missed those thanks

i'll put up a ticket to chase up the `into_inner` boilerplate, it probably needs to be traitified on the wasmer crate side of things, if we want to avoid wrapper functions or macros on this side

i had a look at creating a `ZomeIo` trait in the zome types crate but it ended up making more boilerplate not less

this PR is already over 1k LOC so let's split that out

**Review by freesig — APPROVED** [review]
Looks good!

**Pull request merged**